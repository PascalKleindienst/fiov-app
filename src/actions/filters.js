// =========================================
// FILTERS ACTIONS =========================
// =========================================

// =============== Text Filter ===============
export const SET_TEXT_FILTER = 'SET_TEXT_FILTER';
export const setTextFilter = (text = '') => ({
    type: SET_TEXT_FILTER,
    text
});

// =============== Sort By Date Filter ===============
export const SORT_BY = 'SORT_BY';
export const sortByDate = () => ({
    type: SORT_BY,
    sortBy: 'date'
});

// =============== Sort By Amount Filter ===============
export const sortByAmount = () => ({
    type: SORT_BY,
    sortBy: 'amount'
});

// =============== Start Date Filter ===============
export const SET_START_DATE = 'SET_START_DATE';
export const setStartDate = (startDate) => ({
    type: SET_START_DATE,
    startDate
});

// =============== End Date Filter ===============
export const SET_END_DATE = 'SET_END_DATE';
export const setEndDate = (endDate) => ({
    type: SET_END_DATE,
    endDate
});
