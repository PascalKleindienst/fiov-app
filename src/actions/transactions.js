// ==============================================
// TRANSACTIONS ACTIONS =========================
// ==============================================
import ajax from 'fetchival';

// =============== Add Transaction ===============

// =============== Remove Transaction ===============

// =============== Update Transaction ===============

// =============== Set Transactions ===============
export const SET_TRANSACTIONS = 'SET_TRANSACTIONS';

export const setTransactions = () => {
    return (dispatch) => {
        return ajax('/api/transactions', { mode: 'cors', credentials: 'same-origin' }).get().then((transactions) => {
            dispatch({
                type: SET_TRANSACTIONS,
                transactions
            });
        });
    };
};