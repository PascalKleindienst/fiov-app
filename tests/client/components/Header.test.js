// ==============================================
// HEADER PAGE TEST =============================
// ==============================================
// Mocks
const helpers = require('../../helpers');
import ajax from 'fetchival';
jest.mock('fetchival');

// Dependencies
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import HeaderContainer, { Header } from '../../../src/components/Header';
import { logout } from '../../../src/actions/auth';

const mockStore = configureMockStore([thunk]);

describe('my header component', () => {
    let wrapper, store;

    beforeEach(() => {
        ajax.mockImplementation(() => ({ get: helpers.resolvePromise({ id: 1 }) }));

        store = mockStore({
            auth: { picture: 'foo.png' },
            locale: { languages: ['en'] }
        });
        
        wrapper = shallow(<HeaderContainer store={store} />);
    });

    test('should render correctly when authenticated', () => {
        const wrapper = shallow(<Header logout={() => {}} auth={ {picture: 'foo.png'} } translate={ (msg) => msg } isAuthenticated={true} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render correctly when NOT authenticated', () => {
        const wrapper = shallow(<Header logout={() => {}} translate={ (msg) => msg } isAuthenticated={false} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should call logout on button click', () => {
        const logout = jest.fn();
        const wrapper = shallow(<Header logout={logout} auth={ {picture: 'foo.png'} } translate={ (msg) => msg } isAuthenticated={true} />);
        
        wrapper.find('.logout').simulate('click');
        expect(logout).toHaveBeenCalled();
    });

    test('should contain auth information', () => {
        expect(wrapper.props().auth).toEqual({ picture: 'foo.png' });
    });

    test('should logout when button is clicked', () => {
        return store.dispatch(wrapper.props().logout).then(() => {
            const actions = store.getActions();
            expect(actions[0]).toEqual({ type: 'LOGOUT' });
        });
    });
});
