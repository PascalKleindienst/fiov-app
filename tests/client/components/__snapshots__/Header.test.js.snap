// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my header component should render correctly when NOT authenticated 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/"
      >
        <h1>
          title
        </h1>
      </Link>
      <div
        className="header__nav"
      >
        <Link
          className="header__nav__item"
          replace={false}
          to="/login"
        >
          menu.login
        </Link>
      </div>
    </div>
  </div>
</header>
`;

exports[`my header component should render correctly when authenticated 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          title
        </h1>
      </Link>
      <div
        className="header__nav"
      >
        <NavLink
          activeClassName="header__nav__item--active"
          ariaCurrent="true"
          className="header__nav__item"
          to="/add"
        >
          transactions.add_transaction
        </NavLink>
        <div
          className="profile has-dropdown"
        >
          <img
            alt=""
            className="profile__picture"
            src="foo.png"
          />
          <div
            className="dropdown-menu"
          >
            <div
              className="dropdown-menu__header"
            >
              <strong>
                user.account
              </strong>
            </div>
            <Link
              className="dropdown-menu__item"
              replace={false}
              to="/profile"
            >
              user.profile
            </Link>
            <Link
              className="dropdown-menu__item logout"
              onClick={[Function]}
              replace={false}
              to="#"
            >
              user.logout
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
